### DISCUSSION TITLE
[Question] How to draw all subgraphs by "DrawMorganBits"/"DrawRDKitBits" in case of bit collision

### DISCUSSION THREAD

**Discussion by mohammad-saber** [rdkit/rdkit#4530] [Q&A] [answered]
Hi, thank you for reading my question. 

I would like to visualize subgraphs of a molecule. However, it seems that `DrawMorganBits/DrawRDKitBits` visualize only one subgraph in case of bit collision (not all subgraphs).

For example: 

```
from rdkit import Chem
from rdkit.Chem import AllChem
from rdkit.Chem import Draw

mol = Chem.MolFromSmiles('C(C(=O)O)N')

bi = {}   
fp = AllChem.GetMorganFingerprintAsBitVect(mol, radius=2, bitInfo=bi)
```

Bit [807] is corresponding to two subgraphs. 

However, when I draw subgraphs by:

```
list_bits = [(mol, x, bi) for x in fp.GetOnBits()]   # (mol, bit, bit info)
legends = [str(x) for x in fp.GetOnBits()]
Draw.DrawMorganBits(list_bits, molsPerRow=4,legends=legends)   
```

It only shows one subgraph for bit 807. 

<img width="633" alt="Screen Shot 2021-09-17 at 12 09 59" src="https://user-images.githubusercontent.com/38423900/133718189-61908d6c-95a4-4d26-8a3d-c09c68d3450a.png">

I am aware of parameter `whichExample` in `DrawMorganBit` / `DrawRDKitBit`, but how can I use it in drawing all subgraphs of a molecule?

**Comment by greglandrum** [comment, marked as answer]
Hi @mohammad-saber 
Here's a bit of sample code which does it:
```
list_bits = []
legends = []
for x in fp.GetOnBits():
    for i in range(len(bi[x])):
        list_bits.append((mol,x,bi,i))
        legends.append(str(x))
Draw.DrawMorganBits(list_bits, molsPerRow=4,legends=legends)   
```

One detail: the fact that the same bit is set by multiple atoms in a molecule does not *automatically* mean that there was a collision. It frequently happens that the exact same atom environment appears multiple times in a molecule.
Here's an example of that:
![image](https://user-images.githubusercontent.com/540511/133719871-7485322b-c8c2-49d6-b102-ba29ff633450.png)

**Reply by mohammad-saber** [reply]
Thank you very much for your help.
